File: DSA/Searching/LinearSeach.java
import java.util.Scanner;
class LinearSeach{
	public int linearSearch(int[]a,int data){
		for(int i=0;i<a.length;i++){
			if(data==a[i])return i;
		}
		return -1;
	}
}
class LinearTester{
	public static void main(String[] args){
		int arr[]={3,4,7,2,9,8,1,6,5};
		int data;
		Scanner sc=new Scanner(System.in);
		System.out.println("Enter a number");
		data=sc.nextInt();
		LinearSeach ls=new LinearSeach();
		int result=ls.linearSearch(arr,data);
		if(result==-1){
			System.out.println("Data not found");
		}else{
			System.out.println("Data found at index of" +result);
		}
	}
}

File: linearseach.c
#include<stdio.h>
int linearsearch(int*,int,int);
int main(){
	int element;
	int arr[]={15,7,9,25,8,39,54,79};
	int len=sizeof(arr)/sizeof(arr[0]);
	printf("Enter a element\n");
	scanf("%d",&element);
	int k=linearsearch(arr,len,element);
	if(k == -1){
		printf("element is not found\n");
	}else{
		printf("element is  found at index %d\n",k);
	}
	return 0;
}
int linearsearch(int*a,int len ,int element){
//	int result=-1;
	for(int i=0;i<len;i++){
		if(a[i]==element){
			//result=i;
			//break;
			return i;
		}
	}
	return -1;
}
